libbpf-rs/src/program.rs

```diff
@@ -46,6 +47,29 @@ impl OpenProgram {
         let ret = unsafe { libbpf_sys::bpf_program__set_autoload(self.ptr, autoload) };
         util::parse_ret(ret)
     }
+
+    pub fn set_attach_target(
+        &mut self,
+        attach_prog_fd: i32,
+        attach_func_name: Option<String>,
+    ) -> Result<()> {
+        let ret = if let Some(name) = attach_func_name {
+            // NB: we must hold onto a CString otherwise our pointer dangles
+            let name_c = util::str_to_cstring(&name)?;
```

Comment:
I don't think we need to create a CString for this. We don't need to persist the string inside `bpf_program__set_attach_target`

Reply:
We have to hold this `CString` so that `as_ptr` remain valid.

https://github.com/libbpf/libbpf-rs/blob/42b84c0e4a22d8e398649fe8e787080e8340a2db/libbpf-rs/src/object.rs#L67-L74

Reply:
> We have to hold this CString so that as_ptr remain valid.

Can you elaborate on this ? Inside `bpf_program__set_attach_target`, the name is only used for strcmping and snprintfing so it looks like this pointer does not need to be persisted anywhere. Why do we need to hold onto the CString and what would cause the pointer dangle?

Reply:
The pointer remains valid as long as you hold the CString.

Please refer to https://doc.rust-lang.org/std/ffi/struct.CString.html#examples-1

Reply:
```rust
let ptr = CString::new("Hello").expect("CString::new failed").as_ptr();
// the temporary CString destruct at here, ptr points to freed memory
```

Reply:
Interesting, that seems to directly contradict their documentation about the Cstring being deallocated immediately after the `CString::new("Hello").expect("CString::new failed").as_ptr()` expression is evaluated.

I tested it locally and I verified that the pointer does remain valid, so you're right :) I'll merge the changes in.